source/base/MessageHandler.cpp

```diff
@@ -109,15 +110,27 @@ void MessageHandler::handleMessageReceived(Swift::Message::ref message)
             isGroupMessage = true;
         }
 
+        // xep 0359
```

Comment:
Make the message ID the default one. Only if no message ID can be found, use the StanzaID.

Reply:
Easy, will do!